tekton/utils/pipeline.go

```diff
@@ -78,6 +78,30 @@ func (pr *PipelineRef) GetRevision() (string, error) {
 	return "", fmt.Errorf("no revision found")
 }
 
+// GetPipelineGitRef returns a string containing the url to a Git repo and the path to a Pipeline within that repo.
+// If the git resolver is not used or no url and a pathInRepo params are supplied, the function will fail.
+func (pr *PipelineRef) GetPipelineGitRef() (string, error) {
+	if pr.Resolver != "git" {
+		return "", fmt.Errorf("not a git ref")
+	}
+
+	var url, pathInRepo string
+	for _, param := range pr.Params {
+		switch param.Name {
+		case "url":
+			url = param.Value
+		case "pathInRepo":
+			pathInRepo = param.Value
+		}
+	}
+
+	if url == "" || pathInRepo == "" {
+		return "", fmt.Errorf("unexpected git ref params")
+	}
+
+	return fmt.Sprintf("%s/%s", url, pathInRepo), nil
```

Comment:
Shouldn't this be based on just the `pathInRepo`? Because repo url could be somebody's fork, for example for e2e runs, or occasionally when people point to a fork repo in their RPA (but ideally that shouldn't happen and is pretty rare). So if `url` is included, we would need an entry in the config for each fork.

Reply:
I am not sure we need `url`. but now that I think about it...we'll have an issue when we merge to **development** if we only look at `pathInRepo` ... staging nor production will not have been converted. Which means we can only update `ReleaseServiceConfig` when the change has been promoted all the way to production. Not sure that works either.

Reply:
We talked about this and are hoping to do something like this
```
Stage
*/release-service-catalog.git 	*	pipelines/managed/e2e/e2e.yaml

Prod
https://github.com/konflux-ci/release-service-catalog.git	development	pipelines/managed/e2e/e2e.yaml
https://github.com/konflux-ci/release-service-catalog.git	staging		pipelines/managed/e2e/e2e.yaml
https://github.com/konflux-ci/release-service-catalog.git	development	pipelines/managed/fbc-release/fbc-release.yaml
https://github.com/konflux-ci/release-service-catalog.git	*		pipelines/managed/rh-advisories/rh-advisories.yaml
```
where if a pipeline is listed, it uses emptyDir. We need wildcards, especially in stage, for forks and branches for e2e

Reply:
Ok, this is changed to a list of structs which allows to set the values for url, revision and pathInRepo. All the values are required but both url and revision allows you to use regex expressions so you can set both to `.*` and it will match any value.